### PR TITLE
Api/book crud

### PR DESCRIPTION
**Pull request by juanmbellini**
# Summary

This PR adds the REST layer for Books' CRUD operations, according to the trello card: [https://trello.com/c/Wm9EhU7c](https://trello.com/c/Wm9EhU7c)

# Notes

- Makes use of [Spring Hateoas](https://spring.io/projects/spring-hateoas) Library.
- Remove the greeting controller and view

The base branch is repositories/book, which should be changed to dev once [the repositories/book PR](https://github.com/wolox-training/jmb-java/pull/4) is merged.

# Trello Card

https://trello.com/c/Wm9EhU7c

### CODE REVIEW

File: src/main/java/wolox/training/web/controllers/BookController.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    /**
+     * Constructor.
+     *
+     * @param bookRepository The {@link BookRepository} used to read and write {@link Book} data.
+     */
+    @Autowired
+    public BookController(final BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
```
Comment (Mujics) [inline src/main/java/wolox/training/web/controllers/BookController.java:41]: is there a reason to autowired it in the constructor and not just add the annotations on the attribute?

File: src/main/java/wolox/training/web/controllers/BookController.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    /**
+     * Endpoint for getting a {@link Book} by its {@code id}.
+     *
+     * @param id The id of the {@link Book} to be retrieved.
+     * @return A {@link ResponseEntity} containing the {@link Book} with the given {@code id} if it
+     * exists, or with 404 Not Found {@link ResponseEntity} otherwise.
+     */
+    @GetMapping("/{id}")
+    public ResponseEntity<Book> getById(@PathVariable("id") final long id) {
+        final var book = bookRepository.findById(id);
```
Comment (Mujics) [inline src/main/java/wolox/training/web/controllers/BookController.java]: is the a Java 10 var? or something else?

Comment (juanmbellini) [reply src/main/java/wolox/training/web/controllers/BookController.java]: Yes, Java 10 var. I'm using Java 11 though

File: src/main/java/wolox/training/web/controllers/BookController.java (hunk 1 above)
Comment (juanmbellini) [reply src/main/java/wolox/training/web/controllers/BookController.java:41]: There are several benefits. What I consider the main benefit is that is allows to use final variables.
Here is some info on the following sites: 
- https://stackoverflow.com/a/40620318/5373655
- https://www.vojtechruzicka.com/field-dependency-injection-considered-harmful/
- https://www.javacodegeeks.com/2015/01/dependency-injection-field-vs-constructor-vs-method.html
- https://blog.marcnuri.com/inyeccion-de-campos-desaconsejada-field-injection-not-recommended-spring-ioc/

Comment (Mujics) [reply src/main/java/wolox/training/web/controllers/BookController.java:41]: Excellent, did not know that.  👍

File: src/main/java/wolox/training/web/controllers/BookController.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    /**
+     * Endpoint for getting a {@link Book} by its {@code id}.
+     *
+     * @param id The id of the {@link Book} to be retrieved.
+     * @return A {@link ResponseEntity} containing the {@link Book} with the given {@code id} if it
+     * exists, or with 404 Not Found {@link ResponseEntity} otherwise.
+     */
+    @GetMapping("/{id}")
+    public ResponseEntity<Book> getById(@PathVariable("id") final long id) {
+        return bookRepository.findById(id)
```
Comment (Mujics) [inline src/main/java/wolox/training/web/controllers/BookController.java:65]: love the use of map orElse 👍

**Review by Mujics — APPROVED** [review]

**Review by darkBasic666 — APPROVED** [review]

**Pull request merged**